app/src/main/java/com/sapuseven/untis/activities/LoginDataInputActivity.kt

```diff
@@ -305,6 +306,11 @@ class LoginDataInputActivity : BaseActivity() {
 						?: return@run
 				}
 			val userDataResponse = loadUserData(apiUrl, appSharedSecret) ?: return@run
+			val bookmarks = if (existingUserId == null){
+				ArrayList()
+			} else {
+				userDatabase.getUser(existingUserId!!)!!.bookmarks
+			}
```

Comment:
Untested, but a suggestion to simplify this snippet
```suggestion
			val bookmarks = existingUserId?.let { user -> userDatabase.getUser(user)?.bookmarks } ?: emptyList()
```